homeassistant/components/climate/__init__.py

```diff
@@ -276,128 +302,152 @@ def state_attributes(self) -> dict[str, Any]:
     @property
     def temperature_unit(self) -> str:
         """Return the unit of measurement used by the platform."""
-        raise NotImplementedError()
+        if hasattr(self, "_attr_temperature_unit"):
```

Comment:
We don't need this check if we accept raising AttributeError. Accessing a missing attribute will raise that by default.

Reply:
You are absolutely right, adjusted it.